apps/schematic/api/schematic_api/controllers/schema_controller_impl.py

```diff
@@ -217,118 +250,171 @@ def get_schema_attributes(schema_url: str) -> tuple[Union[str, BasicError], int]
 def get_node_properties_from_schematic(
     node_label: str,
     schema_url: str,
-) -> list[str]:
+) -> list[NodeProperty]:
     """Gets the properties associated with the node
 
     Args:
         schema_url (str): The URL of the schema in jsonld form
         node_label (str): The label of the node
 
     Returns:
-        list[str]: A list of properties of the node
+        list[NodeProperty]: A list of properties of the node
```

Comment:
I am having a little trouble following what is happening here... is NodeProperty an object or class? And you're adding properties to it, then that is in a list?

Reply:
Yeah, NodeProperty is a class, that currently just just contains the name of the node. I noticed its exactly the same as the node class. I think after the schema refactor has been released, but before the API release I'm goign to go through and clean up some of th names of the classes.

Reply:
You're correct this is confusing, I'm going to fix this.